### PR TITLE
fcitx5-skk

### PR DESCRIPTION
**Pull request by ksqsf**
Tests:

- Dict can be loaded (see Debug output)
- type `aiueo` to get `あいうえお`
- type `q` followed by `aiueo` to get `アイウエオ`
- type `Nihongo SPC TAB` to get `日本語`
- type `Hajimemashite SPC Yoroshiku SPC o Negai SPC shimasu` to get `初めまして宜しくお願します`
- type `Koushou SPC SPC` to get a candidate window

test user word:
- type `Kenryuu SPC`, preedit='🔻けんりゅう【】'
- type `Inui SPC`, preedit='🔻けんりゅう【乾】'
- type `Ryuu SPC` and select 隆, preedit='🔻けんりゅう【乾隆】'
- type `RET` to commit
- type `Kenryuu SPC` to get `乾隆` directly

### CODE REVIEW

File: patches/skk-XDG_DATA_DIR.patch
Comment (ksqsf) [inline patches/skk-XDG_DATA_DIR.patch:1]: Maybe I should submit it to the upstream.

File: scripts/skk.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+mkdir -p $dict_dir
+cd $dict_dir
+wget -nc https://skk-dev.github.io/dict/SKK-JISYO.L.gz
+gunzip -f SKK-JISYO.L.gz
+mv SKK-JISYO.L $dict_dir/SKK-JISYO.L
+cd -
+
+# Build fcitx5-skk
+install_deps libxkbcommon pcre2 glib libgee libskk json-glib
+f5m_configure -DENABLE_QT=OFF
```
Comment (ksqsf) [inline scripts/skk.sh]: There is an option `SKK_DEFAULT_PATH` defaulting to `/usr/share/skk/SKK-JISYO.L`, but it is merely a default path used in the Add Dict dialog, and the GUI config tool is not enabled by our build. So keeping it as-is should be harmless, at least for now (when we do not enable GUI config tools).

Ideally, the default path should be `~/Library/fcitx5/share/skk-dict/SKK-JISYO.L`, but that requires code changes similar to the `XDG_DATA_DIR` patch.

**Review by eagleoflqj — CHANGES_REQUESTED** [review]
Screenshot 2024-01-07 at 1 47 32 PM

The plugin and main program are downloaded from CI.
I guess your local version works because it loads something from /tmp/fcitx5? Could you please remove that dir and retest?

**Comment by eagleoflqj** [comment]
Also I'm not able to build it locally.
```
CMake Error in src/CMakeLists.txt:
  Imported target "PkgConfig::GObject2" includes non-existent path

    "/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/ffi"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Comment by ksqsf** [comment]
Reproduced. I'll try to solve it soon.

> ```
>   Imported target "PkgConfig::GObject2" includes non-existent path
> 
>     "/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/ffi"
> ```

You may need to reinstall `pkg-config`. It seems to hardcode MacOS SDK path. (I also had to do this...)

**Review by eagleoflqj — APPROVED** [review]
This is huge considering dependencies. Great job!

**Pull request merged**